owncloudApp/src/main/java/com/owncloud/android/workers/UploadFileFromContentUriWorker.kt

```diff
@@ -134,6 +147,34 @@ class UploadFileFromContentUriWorker(
         }
     }
 
+    private fun copyFileToLocalStorage() {
+        val inputStream = appContext.contentResolver.openInputStream(contentUri)
+
+        val localStorageProvider: LocalStorageProvider by inject()
+        cachePath = localStorageProvider.getTemporalPath(account.name) + uploadPath + UriUtils.getDisplayNameForUri(contentUri, appContext)
+        val cacheFile = File(cachePath)
+        val cacheDir = cacheFile.parentFile
+        if (cacheDir != null) {
+            if (!cacheDir.exists()) {
+                cacheDir.mkdirs()
+            }
+        }
+        cacheFile.createNewFile()
+
+        val outputStream = FileOutputStream(cachePath)
+        val buffer = ByteArray(4096)
+        if (inputStream != null) {
+            var count = inputStream.read(buffer)
+            while (count > 0) {
+                outputStream.write(buffer, 0, count)
+                count = inputStream.read(buffer)
+            }
+        }
+
+        mimeType = cacheFile.extension
+        fileSize = cacheFile.length()
```

Comment:
Not sure about this. If we store the new data into a variable but the worker fails, when it is retried, the mime-type and filesize, and local path would be still the same as the first time even when the file is already in cached storage.

I think that we should persist that data so we don't have issues when retrying failed ones.

An already copied to cache upload but failed, for example because server connection, should:
1. Option: retry as UploadFromFileSystem
2. Option: Detect if the file is already in the cache, and skip moving to caché and go straight to the upload

I would go with the second option. What do you think?